src/components/pages/Authentication/hooks.ts

```diff
@@ -5,32 +5,61 @@ import { useLocation, useNavigate } from "react-router-dom";
 import { useAuth, useCleanUrl } from "@reearth/auth";
 import { useGetTeamsQuery } from "@reearth/gql";
 import { useT } from "@reearth/i18n";
-import { useTeam, useNotification } from "@reearth/state";
+import { useTeam, useNotification, useUserId } from "@reearth/state";
 
 // TODO: move hooks to molecules (page components should be thin)
 export default () => {
   const { isAuthenticated, isLoading, error: authError, logout, login } = useAuth();
   const [error] = useCleanUrl();
   const navigate = useNavigate();
   const location = useLocation();
-  const [currentTeam, setTeam] = useTeam();
+  const [currentWorkspace, setCurrentWorkspace] = useTeam();
+  const [currentUserId, setCurrentUserId] = useUserId();
   const [, setNotification] = useNotification();
   const passwordPolicy = window.REEARTH_CONFIG?.passwordPolicy;
   const t = useT();
 
   const { data, loading } = useGetTeamsQuery({ skip: !isAuthenticated });
-  const teamId = currentTeam?.id || data?.me?.myTeam.id;
+
+  if (isAuthenticated && !currentUserId) {
+    setCurrentUserId(data?.me?.id);
+  }
+  const workspaceId = currentWorkspace?.id || data?.me?.myTeam.id;
```

Comment:
I don't think we really need this since we can just use workspace.id.
But If you think this is necessary, please wrap it in a useMemo

Reply:
I think it is necessary especially first time for each user , Done for wrapping

Reply:
48b47ad76a7e35bbd1b45814258e6e94781b712f